core/primitives/src/errors.rs

```diff
@@ -332,6 +332,48 @@ pub struct ActionError {
     pub kind: ActionErrorKind,
 }
 
+#[derive(
+    BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq, Eq, Deserialize, Serialize, RpcError,
+)]
+pub enum ContractCallError {
+    MethodResolveError(MethodResolveError),
+    CompilationError(CompilationError),
+    ExecutionError(String),
+}
```

Comment:
This is really good structure, self documented of what could be wrong in contract call :+!